src/doc/src/reference/config.md

```diff
@@ -190,8 +190,13 @@ variable.
 
 Environment variables will take precedence over TOML configuration files.
 Currently only integer, boolean, string and some array values are supported to
-be defined by environment variables. Descriptions below indicate which keys
-support environment variables.
+be defined by environment variables. [Descriptions below](#configuration-keys)
+indicate which keys support environment variables. Note that only with explicitly
+showing **"Environment: `ENV_VAR`"** a configuration value can be set in the
+form of environment variable. Those showing **"Environment: not supported"** or
+missing the entire **"Environment:"** field are not able to set via environment
+variables at this moment,
```

Comment:
A challenge in documentation is putting in the right amount.  Being too concise fails but also putting in too much can cause people to be overwhelmed in finding what they are looking for.

Calling what these clearly specified fields and enumerating what the words means, to me, seem like its in the too much category.  I took a glance over the issues and from that glance, I feel like we could shrink this a bit

```
indicate which keys support environment variables and otherwise they are not supported due to [technicial issues](https://github.com/rust-lang/cargo/issues/5416).
```